File: catalogue-application/src/main/java/io/quartic/catalogue/io/quartic/catalogue/datastore/GoogleDatastoreBackendConfig.java
package io.quartic.catalogue.io.quartic.catalogue.datastore;

import io.quartic.catalogue.StorageBackend;
import io.quartic.catalogue.StorageBackendConfig;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public class GoogleDatastoreBackendConfig implements StorageBackendConfig {
    @Valid
    @NotNull
    private String projectId;

    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    @Override
    public StorageBackend build() {
        return GoogleDatastoreBackend.remote(projectId);
    }
}

File: catalogue/src/main/kotlin/io/quartic/catalogue/datastore/GoogleDatastoreBackendConfig.kt
package io.quartic.catalogue.datastore

import io.dropwizard.setup.Environment
import io.quartic.catalogue.StorageBackend
import io.quartic.catalogue.StorageBackendConfig
import io.quartic.common.secrets.SecretsCodec

data class GoogleDatastoreBackendConfig(
    val projectId: String,
    val namespace: String
) : StorageBackendConfig {
    override fun build(env: Environment, secretsCodec: SecretsCodec): StorageBackend = GoogleDatastoreBackend.remote(projectId, namespace)
}
